daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -234,7 +236,11 @@ class CLIConf(arguments: Array[String]) extends scallop.ScallopConf(arguments)
       case "on" => ValidationMode.Full
       case "limited" => ValidationMode.Limited
       case "off" => ValidationMode.Off
-      case _ => throw new Exception("Unrecognized ValidationMode %s.  Must be 'on', 'limited' or 'off'.".format(s))
+      case Validator.MultiArgsPattern(str, args) if Validators.exists(str) =>
+        ValidationMode.Custom(str, args.split(",").map(_.split("=")).map(kv => Validator.Argument(kv.head, kv.last)))
+      case Validator.DefaultArgPattern(str, arg) if Validators.exists(str) => ValidationMode.Custom(str, Seq(Validator.Argument(arg)))
+      case Validator.NoArgsPattern(str) if Validators.exists(str) => ValidationMode.Custom(str, Seq.empty)
```

Comment:
Are these patterns expected to be used via an API user? If not, they should probably only be defined in this Main.scala if the format is specific to the CLI usage.

Reply:
Sounds good.  They are only useful to CLI but wasnt sure where to stash them.

Reply:
moved to cli